nodes/srl/srl.go

```diff
@@ -585,18 +584,7 @@ func (n *srl) addDefaultConfig(ctx context.Context) error {
 			iface.BreakoutNo = ifNameParts[2]
 		}
 
-		// for MACVlan interfaces we need to figure out the parent interface MTU
-		// and specifically define it in the config
-		//
-		// via the endpoint we acquire the link, and check if the link is of type LinkMacVlan
-		// if so cast it and get the parent Interface MTU and finally set that for the interface
-		if link, ok := e.GetLink().(*links.LinkMacVlan); ok {
-			mtu, err := link.GetParentInterfaceMtu()
-			if err != nil {
-				return err
-			}
-			iface.Mtu = mtu
-		}
+		iface.Mtu = e.GetLink().GetMtu()
```

Comment:
this is what causes config to break now in the srl tests in CI

two issues here

1. when we use `srl` node kind the errors from srl postdeploy are not visible (need to understand why)
2. when changed to `nokia_srlinux` kind, the following becomes evident:
```
ERRO[0017] failed to run postdeploy task for node srl1: command execution error:At line 10: Error: Error in path: .interface{.name=="ethernet-1/1"}.mtu
    [InvalidArgument] mtu 9500 out of range [1500-9412]
 
ERRO[0017] failed to run postdeploy task for node srl2: command execution error:At line 13: Error: Error in path: .interface{.name=="ethernet-1/1"}.mtu
    [InvalidArgument] mtu 9500 out of range [1500-9412]
```

the default 9500 MTU shouldn't be used in the srl config, as the 9500 link mtu is a veth mtu only, not for config in NOSes.

Reply:
in [b51def3](https://github.com/srl-labs/containerlab/pull/1532/commits/b51def3673f0a8b73e8ab2ff87eced9ecd5b5689) I changed setting the mtu value in the template endpoints struct only if it is not the default veth mtu value.

Reply:
So here is the issue with SR Linux and low MTU interfaces.
SR Linux emulates BCOM datapath, and port mtu on recent hw platforms can't be less that 1500B.
When using vxlan interfaces we ultimately rely on the MTU of the outgoing (parent) interfaces, which is often configured with 1500B itself.
This leaves vxlan interface with 1450B MTU which is too low, and can't be configured on the SR Linux platform.

Because of that, using vlxan interfaces inside the SRL netns can only be done if the parent interface of the vlxan tunnel has 1550B+ MTU.